test/tests/functional/pbs_maintenance_reservations.py

```diff
@@ -525,7 +539,7 @@ def test_maintenance_reconfirm_reservation_and_run(self):
         rid2 = self.server.submit(r2)
 
         exp_attr = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2'),
-                    'resv_nodes': '(%s:ncpus=1)' % self.momB.shortname}
+                    'resv_nodes': (EQ, '(%s:ncpus=1)' % self.hostB)}
```

Comment:
This is added because, in PTL if reserve_state is not matched in first attempt, then resv_nodes was also checked against reg_expr from second attempt onwards. I had to explicitly give the EQ operator for resv_nodes.
Ex:
2021-06-23 07:11:52,417 INFO     expect on server uv-01: reserve_state ~ RESV_CONFIRMED|2 && resv_nodes = (uv-01[0]:ncpus=1) reservation R159.uv-01  got: reserve_state = RESV_DEGRADED
2021-06-23 07:11:52,918 INFOCLI  uv-01(run_cmd): /opt/pbs/bin/pbs_rstat -f R159.uv-01
2021-06-23 07:11:52,938 INFO     expect on server uv-01: reserve_state ~ RESV_CONFIRMED|2 && resv_nodes ~ (uv-01[0]:ncpus=1) reservation R159.uv-01 attempt: 2  got: reserve_state = RESV_DEGRADED

Reply:
I thought the last time I looked at expect(), it grabbed the operator used in the () and used it for everything.  I might just be thinking about the difference between the operator in () and the op parameter though.  I guess this is working for you though.